python_src/src/lamp_py/postgres/postgres_schema.py

```diff
@@ -17,34 +17,49 @@ class VehicleEvents(SqlBase):  # pylint: disable=too-few-public-methods
 
     __tablename__ = "vehicle_events"
 
-    pk_id = sa.Column(sa.Integer, primary_key=True)
-
-    # hash of trip identifiers
-    trip_hash = sa.Column(
-        sa.LargeBinary(16), nullable=False, index=True, unique=False
-    )
+    # trip identifiers
+    service_date = sa.Column(sa.Integer, nullable=False)
+    pm_trip_id = sa.Column(sa.Integer, nullable=False)
 
     # stop identifiers
-    stop_sequence = sa.Column(sa.SmallInteger, nullable=False)
+    stop_sequence = sa.Column(sa.SmallInteger, nullable=True)
     stop_id = sa.Column(sa.String(60), nullable=False)
     parent_station = sa.Column(sa.String(60), nullable=False)
 
     # stop link fields
     # previous_trip_stop_pk_id = sa.Column(sa.Integer, nullable=True, index=True)
     # next_trip_stop_pk_id = sa.Column(sa.Integer, nullable=True, index=True)
 
-    # hash of trip and stop identifiers
-    trip_stop_hash = sa.Column(
-        sa.LargeBinary(16), nullable=False, index=True, unique=True
-    )
-
     # event timestamps used for metrics
     vp_move_timestamp = sa.Column(sa.Integer, nullable=True)
     vp_stop_timestamp = sa.Column(sa.Integer, nullable=True)
     tu_stop_timestamp = sa.Column(sa.Integer, nullable=True)
 
+    # event metrics values
+    travel_time_seconds = sa.Column(sa.Integer, nullable=True)
+    dwell_time_seconds = sa.Column(sa.Integer, nullable=True)
+    headway_trunk_seconds = sa.Column(sa.Integer, nullable=True)
+    headway_branch_seconds = sa.Column(sa.Integer, nullable=True)
+
     updated_on = sa.Column(sa.TIMESTAMP, server_default=sa.func.now())
 
+    __mapper_args__ = {
+        "primary_key": [
+            service_date,
+            pm_trip_id,
+            parent_station,
+        ]
+    }
+
+
+sa.Index(
+    "ix_vehicle_events_composite_1",
+    VehicleEvents.service_date,
+    VehicleEvents.pm_trip_id,
+    VehicleEvents.parent_station,
+    unique=True,
+)
```

Comment:
I have a feeling this will change during the index/schema ticket. But as part of that PR  I can be sure to add comments on the utility of each index.